pkg/controllers/deprovisioning/suite_test.go

```diff
@@ -2223,43 +2300,44 @@ func fromInt(i int) *intstr.IntOrString {
 	return &v
 }
 
-func ExpectMakeNewNodesReady(ctx context.Context, client client.Client, wg *sync.WaitGroup, numNewNodes int, existingNodes ...*v1.Node) {
-	existingNodeNames := sets.NewString()
-	for _, existing := range existingNodes {
-		existingNodeNames.Insert(existing.Name)
-	}
+func ExpectMakeNewMachinesReady(ctx context.Context, c client.Client, wg *sync.WaitGroup, cluster *state.Cluster,
+	cloudProvider cloudprovider.CloudProvider, numNewMachines int, existingMachines ...*v1alpha5.Machine) {
+
+	existingMachineNames := sets.NewString(lo.Map(existingMachines, func(m *v1alpha5.Machine, _ int) string {
+		return m.Name
+	})...)
+
 	wg.Add(1)
 	go func() {
+		machinesMadeReady := 0
+		ctx, cancel := context.WithTimeout(ctx, time.Second*10) // give up after 10s
 		defer GinkgoRecover()
 		defer wg.Done()
-		start := time.Now()
+		defer cancel()
 		for {
 			select {
 			case <-time.After(50 * time.Millisecond):
-				// give up after 10 seconds
-				if time.Since(start) > 10*time.Second {
-					return
-				}
-				var nodeList v1.NodeList
-				err := client.List(ctx, &nodeList)
-				if err != nil {
+				machineList := &v1alpha5.MachineList{}
+				if err := c.List(ctx, machineList); err != nil {
 					continue
 				}
-				nodesMadeReady := 0
-				for i := range nodeList.Items {
-					n := &nodeList.Items[i]
-					if existingNodeNames.Has(n.Name) {
+				for i := range machineList.Items {
+					m := &machineList.Items[i]
+					if existingMachineNames.Has(m.Name) {
 						continue
 					}
-					ExpectMakeNodesReady(ctx, env.Client, n)
-					nodesMadeReady++
-					// did we make all of the nodes ready that we expected?
-					if nodesMadeReady == numNewNodes {
+					ExpectMachineDeployedWithOffset(1, ctx, c, cluster, cloudProvider, m)
+					ExpectMakeMachinesReadyWithOffset(1, ctx, c, m)
+
+					machinesMadeReady++
+					existingMachineNames.Insert(m.Name)
+					// did we make all the nodes ready that we expected?
+					if machinesMadeReady == numNewMachines {
 						return
 					}
 				}
 			case <-ctx.Done():
-				return
+				Fail(fmt.Sprintf("waiting for machines to be ready, %s", ctx.Err()))
```

Comment:
nit: Should say something about context cancelled here.

Reply:
context cancelled is returned by ctx.Err()